### ISSUE TITLE
Problem in Quick, Draw! (Python)

### ISSUE THREAD

**Issue by AnshDMishra** [Puliczek/github-achievements-academy#2387]
Problem in Quick, Draw! (Python)
The Quick, Draw! dataset contains millions of doodles collected from an online game. The challenge in Python typically involves:

Reading and processing the .ndjson (Newline Delimited JSON) dataset
Converting stroke-based data into images
Building models for recognition (e.g., using TensorFlow/Keras)
Common Issues
Parsing .ndjson files

The dataset is stored in .ndjson, where each line is a JSON object.
Incorrect parsing leads to issues while loading the data.
Preprocessing Stroke Data into Images

The dataset stores strokes as a sequence of points, not as a pixel-based image.
Transforming stroke data into a proper format (e.g., numpy arrays) is necessary.
Model Submission Issues

If submitting to a competition (e.g., Kaggle), the format must match the expected output.
Model predictions should match the required structure.

**Comment by AnshDMishra** [comment]
import ndjson
import numpy as np
import pandas as pd

# Load the dataset
file_path = "full_simplified_cat.ndjson"  # Example for 'cat' class
with open(file_path) as f:
    data = ndjson.load(f)

# Convert to a DataFrame
df = pd.DataFrame(data)

# Display some sample strokes
df.head()



import matplotlib.pyplot as plt

def draw_stroke(strokes):
    plt.figure(figsize=(2, 2))
    for stroke in strokes:
        plt.plot(stroke[0], -np.array(stroke[1]), 'k')
    plt.axis('off')
    plt.show()

draw_stroke(df.iloc[0]['drawing'])



submission = pd.DataFrame({
    'key_id': df['key_id'],
    'word': 'cat'  # Replace with actual predictions
})

submission.to_csv('submission.csv', index=False)
